### PR TITLE
JSDoc-comments# added JSDoc comments

### PR DESCRIPTION
**Pull request by Amar-NHS**
## Description

Added JS Docs comments on the Lambda files.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/authenticatorLambda/authenticatorLambda.js [hunk 1]
```diff
@@ -91,7 +101,18 @@ export const handler = async (event) => {
...
+ * Generates a session for API Gateway Lock-down.
+ *
+ * @function generateAPIGatewayLockdownSession
+ * @async
+ * @param {string} environment - The environment name.
+ * @param {Object} client - The DynamoDB client.
+ * @param {string} apiSessionId - The API session ID.
+ * @param {string} userId - The user ID.
+ * @param {Function} updateAPISessionTable - The function to update the API session table.
+ */
```
Comment (chenglawson1) [inline terraform/src/authenticatorLambda/authenticatorLambda.js:115]: Could add @throws.

File: terraform/src/authenticatorLambda/authenticatorLambda.js [hunk 2]
```diff
@@ -220,7 +268,16 @@ export async function getCIS2SignedJWT(
...
 }
 
-// Function to get Secrets
+/**
+ * Retrieves 'Secret value' from secrets manager by passing in 'Secret name' from AWS Secrets Manager
+ *
+ * @function getSecret
+ * @async
+ * @param {string} secretName - Name of the secret to retrieve
+ * @param {Object} client - Secrets Manager client
```
Comment (chenglawson1) [inline terraform/src/authenticatorLambda/authenticatorLambda.js:277]: Could change type Object to SecretsManagerClient.

File: terraform/src/authenticatorLambda/authenticatorLambda.js [hunk 3]
```diff
@@ -357,7 +457,20 @@ export async function getUserRole(uuid) {
...
+ *
+ * @function checkAuthorization
+ * @async
+ * @param {Object} userAuthData - The user authentication data.
+ * @param {Object} tokens - The tokens object.
+ * @param {string} activityCode - The required activity code for authorization.
+ * @param {string} clientID - The client ID.
+ * @param {Function} extractClaims - The function to extract claims from the ID token.
+ * @param {Function} validateTokenExpiration - The function to validate token expiration with auth time.
+ * @param {Function} validateTokenSignature - The function to validate the token signature.
```
Comment (chenglawson1) [inline terraform/src/authenticatorLambda/authenticatorLambda.js:471]: Are the parameter names in the comment intentionally different from the actuals?

File: terraform/src/authenticatorLambda/authenticatorLambda.js [hunk 4]
```diff
@@ -433,7 +553,15 @@ export async function extractClaims(idToken) {
...
 }
 
-// Function to validate the expiration time (exp claim)
+/**
+ * Validates the expiration of the token with the authentication time(exp claim).
+ *
+ * @function validateTokenExpirationWithAuthTime
+ * @async
+ * @param {string} idToken - The ID token.
+ * @param {number} authTime - The authentication time.
```
Comment (chenglawson1) [inline terraform/src/authenticatorLambda/authenticatorLambda.js:562]: Number of parameters in the comment is different to actual, and idToken is different from actual?

File: terraform/src/authenticatorLambda/authenticatorLambda.js [hunk 5]
```diff
@@ -449,7 +577,15 @@ export async function validateTokenExpirationWithAuthTime(token) {
...
 }
 
-// Function to validate the signature of id token received
+/**
+ * Validates the signature of the id token received.
+ *
+ * @function validateTokenSignature
+ * @async
+ * @param {string} idToken - The ID token.
+ * @returns {boolean} True if the signature is valid, otherwise false.
```
Comment (chenglawson1) [inline terraform/src/authenticatorLambda/authenticatorLambda.js:586]: Need to include jwksUri.

File: terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js [hunk 6]
```diff
@@ -96,7 +106,18 @@ export const handler = async (event, context) => {
...
-// Retrieve and Parse the JSON file
+/**
+ * This function retrieves and parse JSON file from S3 bucket
+ *
+ * @function retrieveAndParseJSON
+ * @async
+ * @param {Function} getJSONFunc - Function to get JSON from S3
+ * @param {string} bucket - S3 bucket name
+ * @param {string} key - S3 object key
+ * @param {Object} client - S3 client
```
Comment (chenglawson1) [inline terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js:117]: Could change Object to S3Client.

File: terraform/src/sendTestResultErrorAckQueueLambda/sendTestResultErrorAckQueueLambda.js [hunk 7]
```diff
@@ -43,7 +50,18 @@ export const handler = async (event) => {
...
-// Function to send Message to SQS Queue
+
+/**
+ * Sends a message to the SQS Queue.
+ *
+ * @function sendMessageToQueue
+ * @async
+ * @param {Object} message - The message object to be sent to the SQS queue.
+ * @param {string} queue - The SQS queue URL.
+ * @param {Object} sqsClient - The SQS client.
```
Comment (chenglawson1) [inline terraform/src/sendTestResultErrorAckQueueLambda/sendTestResultErrorAckQueueLambda.js:61]: Could change Object to SQSClient.
Same with S3Client.

File: terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js [hunk 8]
```diff
@@ -27,7 +33,18 @@ export const handler = async (event) => {
...
-// Function to process the message published on SNS Topic and send Message to SQS Queue
+
+/**
+ * process the message published on SNS Topic and sends a message to the SQS Queue.
+ *
+ * @function processSnsMessage
+ * @async
+ * @param {Object} message - The message object to be sent to the SQS queue.
+ * @param {string} queue - The SQS queue URL.
+ * @param {Object} sqsClient - The SQS client.
```
Comment (chenglawson1) [inline terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js:44]: Could change Object to class of client.
Same with S3 client in following.

File: terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js [hunk 9]
```diff
@@ -27,7 +33,18 @@ export const handler = async (event) => {
...
+ * @param {string} snsMessageId - The SNS message ID.
+ * @throws {Error}  Will throw an error if sending message to SQS fails.
```
Comment (chenglawson1) [inline terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js:46]: There seems to be some mismatch in the parameters in the comment and actuals?

File: terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js [hunk 10]
```diff
@@ -10,6 +10,14 @@ const s3 = new S3Client();
 
 const ENVIRONMENT = process.env.ENVIRONMENT;
 
+/**
+ * Lambda handler function to process CSV files from S3 and perform validation checks.
+ *
+ * @function handler
+ * @async
+ * @param {Object} event - The Lambda event object containing S3 event details.
+ * @throws {Error}
```
Comment (chenglawson1) [inline terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js:19]: Handler returns a string when successful.

File: terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js [hunk 11]
```diff
@@ -111,6 +119,17 @@ export const handler = async (event) => {
...
 };
 
+/**
+ * Retrieves a CSV file from S3.
+ *
+ * @function readCsvFromS3
+ * @async
+ * @param {string} bucketName - The name of the S3 bucket.
+ * @param {string} key - The key of the S3 object.
+ * @param {Object} client - The S3 client.
```
Comment (chenglawson1) [inline terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js:129]: Could change Object to s3 client class.

File: terraform/src/authenticatorLambda/authenticatorLambda.js (hunk 1 above)
Comment (Amar-NHS) [reply terraform/src/authenticatorLambda/authenticatorLambda.js:116]: Added those in. thanks

File: terraform/src/authenticatorLambda/authenticatorLambda.js (hunk 2 above)
Comment (Amar-NHS) [reply terraform/src/authenticatorLambda/authenticatorLambda.js]: Updated to SecretsManagerClient. thanks

File: terraform/src/authenticatorLambda/authenticatorLambda.js (hunk 3 above)
Comment (Amar-NHS) [reply terraform/src/authenticatorLambda/authenticatorLambda.js]: Left it by mistake. Corrected now. Thanks!

File: terraform/src/authenticatorLambda/authenticatorLambda.js (hunk 4 above)
Comment (Amar-NHS) [reply terraform/src/authenticatorLambda/authenticatorLambda.js]: Left it by mistake. Corrected now. Thanks!

File: terraform/src/authenticatorLambda/authenticatorLambda.js (hunk 5 above)
Comment (Amar-NHS) [reply terraform/src/authenticatorLambda/authenticatorLambda.js:587]: Added now. Thanks!

File: terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js (hunk 6 above)
Comment (Amar-NHS) [reply terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js]: Updated now. Thanks!

File: terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js [hunk 12]
```diff
@@ -211,7 +291,17 @@ export async function deleteObjectFromS3(bucketName, objectKey, client) {
...
-//Send Message based on the MailBox ID from the config
+/**
+ * Sends an uncompressed message to the MESH mailbox,
+ * based on the MailBox ID from the config.
+ *
+ * @function sendUncompressed
+ * @async
+ * @param {Object} config - Configuration object
+ * @param {Object} msg - Message object
+ * @returns {number} HTTP status code of the response
```
[the remark this answers is not in the document]
Comment (Amar-NHS) [reply terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js:303]: Corrected now. Thanks!

File: terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js [hunk 13]
```diff
@@ -274,6 +373,16 @@ export const getSecret = async (secretName, client) => {
...
 };
 
+/**
+ * Reads a secret from AWS Secrets Manager
+ *
+ * @function readSecret
+ * @async
+ * @param {Function} fetchSecret - Function to get the secret
+ * @param {string} secretName - Name of the secret to retrieve
+ * @param {Object} client - Secrets Manager client
```
[the remark this answers is not in the document]
Comment (Amar-NHS) [reply terraform/src/sendGTMSInvitationBatchLambda/sendGTMSInvitationBatchLambda.js:384]: Updated now. Thanks!

File: terraform/src/sendTestResultErrorAckQueueLambda/sendTestResultErrorAckQueueLambda.js (hunk 7 above)
Comment (Amar-NHS) [reply terraform/src/sendTestResultErrorAckQueueLambda/sendTestResultErrorAckQueueLambda.js]: Updated now. Thanks!

File: terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js (hunk 8 above)
Comment (Amar-NHS) [reply terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js]: Updated now. Thanks!

File: terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js (hunk 9 above)
Comment (Amar-NHS) [reply terraform/src/sendTestResultOkAckQueueLambda/sendTestResultOkAckQueueLambda.js:47]: Corrected now. Thanks!

File: terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js (hunk 10 above)
Comment (Amar-NHS) [reply terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js:20]: Added the @returns. thanks!

File: terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js (hunk 11 above)
Comment (Amar-NHS) [reply terraform/src/validateCaasFeedLambda/validateCaasFeedLambda.js:130]: Updated it. Thanks.

**Review by chenglawson1 — APPROVED** [review]

**Review by Abdul-NHS — APPROVED** [review]

**Pull request merged**